src/Adapter/Cart/AbstractCartHandler.php

```diff
@@ -61,4 +69,38 @@ protected function getCart(CartId $cartId)
 
         return $cart;
     }
+
+    /**
+     * @param ContextStateManager $contextStateManager
+     * @param Cart $cart
+     */
+    protected function setCartContext(ContextStateManager $contextStateManager, Cart $cart): void
+    {
+        $contextStateManager
+            ->saveCurrentContext()
+            ->setCart($cart)
+            ->setCustomer(new Customer($cart->id_customer))
+            ->setCurrency(new Currency($cart->id_currency))
+            ->setLanguage(new Language($cart->id_lang))
+            ->setCountry($this->getCartTaxCountry($cart))
+            ->setShop(new Shop($cart->id_shop))
+        ;
+    }
+
+    /**
+     * @param Cart $cart
+     *
+     * @return Country
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function getCartTaxCountry(Cart $cart): Country
```

Comment:
@Progi1984 

in `Cart` class there's a method doing exactly same thing, `Cart->getTaxCountry`

is there any specific reason to not use it?

Reply:
Indeed, I think we can use `Cart::getTaxCountry`

Reply:
Fixed